cpp/test/dist_sort_test.cpp

```diff
@@ -0,0 +1,287 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cylon/table.hpp>
+
+#include "common/test_header.hpp"
+#include "test_utils.hpp"
+
+namespace cylon {
+void testDistSort(const std::vector<int>& sort_cols,
+                  const std::vector<bool>& sort_order,
+                  std::shared_ptr<Table>& global_table,
+                  std::shared_ptr<Table>& table) {
+  std::shared_ptr<Table> out;
+  auto ctx = table->GetContext();
+  std::shared_ptr<arrow::Table> arrow_output;
+
+  auto status = DistributedSort(table, sort_cols, out, sort_order,
+                                {0, 0, SortOptions::INITIAL_SAMPLE});
+  REQUIRE(status.is_ok());
+  std::shared_ptr<Table> out2;
+  bool eq;
+
+  if (RANK == 0) {
+    status = Sort(global_table, sort_cols, out2, sort_order);
+  } else {
+    auto pool = cylon::ToArrowPool(ctx);
+    std::shared_ptr<arrow::Table> arrow_empty_table;
+    auto arrow_status = util::CreateEmptyTable(
+        global_table->get_table()->schema(), &arrow_empty_table, pool);
+    out2 = std::make_shared<Table>(ctx, arrow_empty_table);
+  }
+  std::shared_ptr<Table> out3;
+  status = Repartition(out2, &out3);
+  status = DistributedEquals(out3, out, eq);
+  REQUIRE(eq);
+}
+
+namespace test {
+TEST_CASE("Dist sort testing", "[dist sort]") {
```

Comment:
Let's parameterize this test case (let's parameterize field `a` for `ArrowNumericTypes`). 

Take a look at this 
https://github.com/cylondata/cylon/blob/2759a303b5f9d601229bc5c8321586e06f072495/cpp/test/sync_comms_test.cpp#L231
This is Catch2 docs on templated testcases
https://github.com/catchorg/Catch2/blob/devel/docs/test-cases-and-sections.md#type-parametrised-test-cases